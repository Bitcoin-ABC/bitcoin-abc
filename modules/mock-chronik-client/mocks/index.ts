// Copyright (c) 2023 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import {
    Block,
    BlockchainInfo,
    Tx,
    TokenInfo,
    ScriptUtxo,
    Utxo,
} from 'chronik-client';

interface ChronikMocks {
    tipHeight: number;
    tipHash: string;
    block: Block;
    blockchainInfo: BlockchainInfo;
    txid: string;
    tx: Tx;
    tokenId: string;
    token: TokenInfo;
    rawTx: string;
    scriptUtxo: ScriptUtxo;
    utxo: Utxo;
}
const DUMMY_SCRIPT_UTXO: ScriptUtxo = {
    outpoint: { txid: '00'.repeat(32), outIdx: 0 },
    blockHeight: 800000,
    isCoinbase: false,
    value: 546,
    isFinal: true,
};
const DUMMY_UTXO: Utxo = {
    outpoint: { txid: '00'.repeat(32), outIdx: 0 },
    script: `76a914${'00'.repeat(20)}88ac`,
    blockHeight: 800000,
    isCoinbase: false,
    value: 546,
    isFinal: true,
};
const chronikMocks: ChronikMocks = {
    tipHeight: 800000,
    tipHash: '0000000000000000115e051672e3d4a6c523598594825a1194862937941296fe',
    block: {
        blockInfo: {
            hash: '0000000000000000115e051672e3d4a6c523598594825a1194862937941296fe',
            prevHash:
                '0000000000000000023ab587190a05e44cdb76fe1dd7949f6187f1e632d2e123',
            height: 800000,
            nBits: 403943060,
            timestamp: 1688808780,
            isFinal: true,
            blockSize: 3094,
            numTxs: 9,
            numInputs: 13,
            numOutputs: 26,
            sumInputSats: 859760862,
            sumCoinbaseOutputSats: 625005728,
            sumNormalOutputSats: 859755134,
            sumBurnedSats: 0,
        },
    },
    blockchainInfo: {
        tipHash:
            '0000000000000000115e051672e3d4a6c523598594825a1194862937941296fe',
        tipHeight: 800000,
    },
    txid: '2c1540131fb7611b72ec3ca9a215ac5c408248327d3c5b5ff987cd2a959ee1d2',
    tx: {
        txid: '2c1540131fb7611b72ec3ca9a215ac5c408248327d3c5b5ff987cd2a959ee1d2',
        version: 2,
        inputs: [
            {
                prevOut: {
                    txid: '1cb37ece9d9dbc19ef2c6d57766ca3b91bdc3e56d203aa0fc1b41a7268c0c9d2',
                    outIdx: 2,
                },
                inputScript:
                    '41ebe634094888c4215b8690546b74695a048a9c49c8a5c6efedd940591ec9039e91cf2a6f7836107afbc41bb6e13fb22d99739aeadf1a740f75efea177cfcd86641210353f81d61d41d6e22c73ab449476113dea124afe3972991cd237e654f15950b7c',
                value: 546,
                sequenceNo: 4294967295,
                token: {
                    tokenId:
                        'aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb1',
                    tokenType: {
                        protocol: 'SLP',
                        type: 'SLP_TOKEN_TYPE_FUNGIBLE',
                        number: 1,
                    },
                    amount: '455320000',
                    isMintBaton: false,
                    entryIdx: 0,
                },
                outputScript:
                    '76a914821407ac2993f8684227004f4086082f3f801da788ac',
            },
            {
                prevOut: {
                    txid: '1cb37ece9d9dbc19ef2c6d57766ca3b91bdc3e56d203aa0fc1b41a7268c0c9d2',
                    outIdx: 3,
                },
                inputScript:
                    '412e7de41834aaf15d16347a59c954cb886d8ec00584bb566252e16f8e1779ce2e4d6ddd91782bd6ba2f695a28ce4115cd0561d27c1ae16b3d8bf77f8186235b1441210353f81d61d41d6e22c73ab449476113dea124afe3972991cd237e654f15950b7c',
                value: 94518653,
                sequenceNo: 4294967295,
                outputScript:
                    '76a914821407ac2993f8684227004f4086082f3f801da788ac',
            },
        ],
        outputs: [
            {
                value: 0,
                outputScript:
                    '6a04534c500001010453454e4420aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb108000000000000271008000000001b237ab0',
            },
            {
                value: 546,
                outputScript:
                    '76a914a805a320360fa685f83605d8e56de6f9d8a7a99988ac',
                token: {
                    tokenId:
                        'aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb1',
                    tokenType: {
                        protocol: 'SLP',
                        type: 'SLP_TOKEN_TYPE_FUNGIBLE',
                        number: 1,
                    },
                    amount: '10000',
                    isMintBaton: false,
                    entryIdx: 0,
                },
            },
            {
                value: 546,
                outputScript:
                    '76a914821407ac2993f8684227004f4086082f3f801da788ac',
                token: {
                    tokenId:
                        'aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb1',
                    tokenType: {
                        protocol: 'SLP',
                        type: 'SLP_TOKEN_TYPE_FUNGIBLE',
                        number: 1,
                    },
                    amount: '455310000',
                    isMintBaton: false,
                    entryIdx: 0,
                },
                spentBy: {
                    txid: 'b334ad4ee98718707142a7a951163f1cb94817269b55ac840a7b416ce256eecd',
                    outIdx: 0,
                },
            },
            {
                value: 94517640,
                outputScript:
                    '76a914821407ac2993f8684227004f4086082f3f801da788ac',
                spentBy: {
                    txid: 'b334ad4ee98718707142a7a951163f1cb94817269b55ac840a7b416ce256eecd',
                    outIdx: 1,
                },
            },
        ],
        lockTime: 0,
        timeFirstSeen: 0,
        size: 467,
        isCoinbase: false,
        tokenEntries: [
            {
                tokenId:
                    'aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb1',
                tokenType: {
                    protocol: 'SLP',
                    type: 'SLP_TOKEN_TYPE_FUNGIBLE',
                    number: 1,
                },
                txType: 'SEND',
                isInvalid: false,
                burnSummary: '',
                failedColorings: [],
                actualBurnAmount: '0',
                intentionalBurn: '0',
                burnsMintBatons: false,
            },
        ],
        tokenFailedParsings: [],
        tokenStatus: 'TOKEN_STATUS_NORMAL',
        isFinal: false,
        block: {
            height: 866581,
            hash: '000000000000000000860f542fac8cbb10614e40ab0f37f90ec4d41b126eb5d9',
            timestamp: 1728930658,
        },
    },
    tokenId: 'aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb1',
    token: {
        tokenId:
            'aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb1',
        tokenType: {
            protocol: 'SLP',
            type: 'SLP_TOKEN_TYPE_FUNGIBLE',
            number: 1,
        },
        timeFirstSeen: 1711776546,
        genesisInfo: {
            tokenTicker: 'CACHET',
            tokenName: 'Cachet',
            url: 'https://cashtab.com/',
            decimals: 2,
            hash: '',
        },
        block: {
            height: 838192,
            hash: '0000000000000000132232769161d6211f7e6e20cf63b26e5148890aacd26962',
            timestamp: 1711779364,
        },
    },
    rawTx: '0200000002d2c9c068721ab4c10faa03d2563edc1bb9a36c76576d2cef19bc9d9dce7eb31c020000006441ebe634094888c4215b8690546b74695a048a9c49c8a5c6efedd940591ec9039e91cf2a6f7836107afbc41bb6e13fb22d99739aeadf1a740f75efea177cfcd86641210353f81d61d41d6e22c73ab449476113dea124afe3972991cd237e654f15950b7cffffffffd2c9c068721ab4c10faa03d2563edc1bb9a36c76576d2cef19bc9d9dce7eb31c0300000064412e7de41834aaf15d16347a59c954cb886d8ec00584bb566252e16f8e1779ce2e4d6ddd91782bd6ba2f695a28ce4115cd0561d27c1ae16b3d8bf77f8186235b1441210353f81d61d41d6e22c73ab449476113dea124afe3972991cd237e654f15950b7cffffffff040000000000000000406a04534c500001010453454e4420aed861a31b96934b88c0252ede135cb9700d7649f69191235087a3030e553cb108000000000000271008000000001b237ab022020000000000001976a914a805a320360fa685f83605d8e56de6f9d8a7a99988ac22020000000000001976a914821407ac2993f8684227004f4086082f3f801da788ac8839a205000000001976a914821407ac2993f8684227004f4086082f3f801da788ac00000000',
    scriptUtxo: DUMMY_SCRIPT_UTXO,
    utxo: DUMMY_UTXO,
};

export default chronikMocks;
