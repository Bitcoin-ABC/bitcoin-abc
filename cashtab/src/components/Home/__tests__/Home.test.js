// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import React from 'react';
import { walletWithXecAndTokens } from 'components/App/fixtures/mocks';
import { walletWithZeroBalanceZeroHistory } from 'components/Home/fixtures/mocks';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'fake-indexeddb/auto';
import localforage from 'localforage';
import { when } from 'jest-when';
import appConfig from 'config/app';
import {
    initializeCashtabStateForTests,
    clearLocalForage,
} from 'components/App/fixtures/helpers';
import CashtabTestWrapper from 'components/App/fixtures/CashtabTestWrapper';

// https://stackoverflow.com/questions/39830580/jest-test-fails-typeerror-window-matchmedia-is-not-a-function
Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
    })),
});

// https://stackoverflow.com/questions/64813447/cannot-read-property-addlistener-of-undefined-react-testing-library
window.matchMedia = query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
});

// Mock a valid sideshift object in window
window.sideshift = {
    show: jest.fn(),
    hide: jest.fn(),
    addEventListener: jest.fn(),
};

describe('<Home />', () => {
    beforeEach(() => {
        // Mock the fetch call for Cashtab's price API
        global.fetch = jest.fn();
        const fiatCode = 'usd'; // Use usd until you mock getting settings from localforage
        const cryptoId = appConfig.coingeckoId;
        // Keep this in the code, because different URLs will have different outputs requiring different parsing
        const priceApiUrl = `https://api.coingecko.com/api/v3/simple/price?ids=${cryptoId}&vs_currencies=${fiatCode}&include_last_updated_at=true`;
        const xecPrice = 0.00003;
        const priceResponse = {
            ecash: {
                usd: xecPrice,
                last_updated_at: 1706644626,
            },
        };
        when(fetch)
            .calledWith(priceApiUrl)
            .mockResolvedValue({
                json: () => Promise.resolve(priceResponse),
            });
    });
    afterEach(async () => {
        jest.clearAllMocks();
        await clearLocalForage(localforage);
    });
    it('Renders the loading component while loading, then the Home screen', async () => {
        // localforage defaults
        const mockedChronik = await initializeCashtabStateForTests(
            walletWithXecAndTokens,
            localforage,
        );
        render(<CashtabTestWrapper chronik={mockedChronik} />);

        // Initially, Home component is not rendered
        expect(screen.queryByTestId('tx-history')).not.toBeInTheDocument();

        // Initially, Loading ctn is rendered
        expect(screen.getByTitle('Cashtab Loading')).toBeInTheDocument();

        // After wallet loads, this is reversed
        await waitFor(() =>
            expect(
                screen.queryByTitle('Cashtab Loading'),
            ).not.toBeInTheDocument(),
        );

        // Wait for the balance to render
        expect(await screen.findByText('9,513.12 XEC')).toBeInTheDocument();

        // The home screen is in the document
        expect(await screen.findByTestId('tx-history')).toBeInTheDocument();

        // No API Error
        await waitFor(() =>
            expect(
                screen.queryByText('Error in chronik connection'),
            ).not.toBeInTheDocument(),
        );
    });
    it('Renders the Home screen with API error', async () => {
        // localforage defaults and API error set
        const mockedChronik = await initializeCashtabStateForTests(
            walletWithXecAndTokens,
            localforage,
            true, // apiError bool
        );
        render(<CashtabTestWrapper chronik={mockedChronik} />);

        // Wait for the app to load
        await waitFor(() =>
            expect(
                screen.queryByTitle('Cashtab Loading'),
            ).not.toBeInTheDocument(),
        );

        // API Error is rendered
        expect(
            await screen.findByText('Error in chronik connection'),
        ).toBeInTheDocument();
    });
    it('Renders backup warning and QR Code if user loads with a new wallet', async () => {
        // localforage defaults
        const mockedChronik = await initializeCashtabStateForTests(
            walletWithZeroBalanceZeroHistory,
            localforage,
        );
        render(<CashtabTestWrapper chronik={mockedChronik} />);

        // Wait for the component to finish loading
        await waitFor(() =>
            expect(
                screen.queryByTitle('Cashtab Loading'),
            ).not.toBeInTheDocument(),
        );

        // Backup warning is rendered
        await screen.findByText('Backup your wallet');
        await screen.findByText(
            'Write down your 12-word seed and keep it in a safe place.',
        );
        await screen.findByText('Do not share your backup with anyone.');

        // Receive QR code is rendered
        expect(screen.getByTitle('Receive')).toBeInTheDocument();
    });
});
