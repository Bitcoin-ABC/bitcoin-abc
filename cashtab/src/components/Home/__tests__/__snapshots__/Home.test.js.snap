// Jest Snapshot v1, https://goo.gl/fbAQLP @generated

exports[`Wallet with BCH balances 1`] = `
Array [
  <div
    className="sc-elJkPf KqQMk"
  >
    <div
      className="sc-eXEjpC fRnTPw"
    >
      <h4
        className="sc-ibxdXY gNKGGy"
      >
        MigrationTestAlpha
      </h4>
      <a
        href="/configure"
        onClick={[Function]}
      >
        <svg
          className="sc-jTzLTM bTdWCF"
        >
          edit.svg
        </svg>
      </a>
      <div>
         
        <button
          aria-checked={false}
          className="ant-switch ant-switch-small"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="switch"
          type="button"
        >
          <div
            className="ant-switch-handle"
          />
          <span
            className="ant-switch-inner"
          >
            <svg
              className="sc-kAzzGY fUTkYj"
            >
              eye-invisible.svg
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      className="sc-kTUwUJ bWNEPO"
    >
      <span
        className="sc-kfGgVZ ezlXcw"
      >
         
        0
         
        XEC
         
      </span>
    </div>
  </div>,
  <div
    className="sc-lhVmIH gJoLAm"
  >
    <div
      className="sc-RefOD bFUtiM"
    >
      <button
        className="sc-iQKALj bpbBVm"
        onClick={[Function]}
      >
        Transactions
      </button>
      <button
        className="sc-iQKALj jssRHA"
        onClick={[Function]}
      >
        eTokens
      </button>
    </div>
    <div
      className="sc-bwCtUz jWzyRI"
    >
      <div />
      <span
        aria-label="party emoji"
        role="img"
      >
        ðŸŽ‰
      </span>
      Congratulations on your new wallet!
       
      <span
        aria-label="party emoji"
        role="img"
      >
        ðŸŽ‰
      </span>
      <br />
       Start using the wallet immediately to receive
       
      XEC
       payments, or load it up with
       
      XEC
       to send to others
      <br />
      <br />
    </div>
    <div
      className="sc-bwCtUz hyFPDq"
    >
      <a
        className="sc-hXRMBi kpUYV"
        href="/tokens"
        onClick={[Function]}
      >
        Create eToken
      </a>
      <p>
        Tokens sent to your 
        eToken
         address will appear here
      </p>
    </div>
  </div>,
]
`;

exports[`Wallet with BCH balances and tokens 1`] = `
Array [
  <div
    className="sc-elJkPf KqQMk"
  >
    <div
      className="sc-eXEjpC fRnTPw"
    >
      <h4
        className="sc-ibxdXY gNKGGy"
      >
        MigrationTestAlpha
      </h4>
      <a
        href="/configure"
        onClick={[Function]}
      >
        <svg
          className="sc-jTzLTM bTdWCF"
        >
          edit.svg
        </svg>
      </a>
      <div>
         
        <button
          aria-checked={false}
          className="ant-switch ant-switch-small"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="switch"
          type="button"
        >
          <div
            className="ant-switch-handle"
          />
          <span
            className="ant-switch-inner"
          >
            <svg
              className="sc-kAzzGY fUTkYj"
            >
              eye-invisible.svg
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      className="sc-kTUwUJ bWNEPO"
    >
      <span
        className="sc-kfGgVZ ezlXcw"
      >
         
        0
         
        XEC
         
      </span>
    </div>
  </div>,
  <div
    className="sc-lhVmIH gJoLAm"
  >
    <div
      className="sc-RefOD bFUtiM"
    >
      <button
        className="sc-iQKALj bpbBVm"
        onClick={[Function]}
      >
        Transactions
      </button>
      <button
        className="sc-iQKALj jssRHA"
        onClick={[Function]}
      >
        eTokens
      </button>
    </div>
    <div
      className="sc-bwCtUz jWzyRI"
    >
      <div />
      <span
        aria-label="party emoji"
        role="img"
      >
        ðŸŽ‰
      </span>
      Congratulations on your new wallet!
       
      <span
        aria-label="party emoji"
        role="img"
      >
        ðŸŽ‰
      </span>
      <br />
       Start using the wallet immediately to receive
       
      XEC
       payments, or load it up with
       
      XEC
       to send to others
      <br />
      <br />
    </div>
    <div
      className="sc-bwCtUz hyFPDq"
    >
      <a
        className="sc-hXRMBi kpUYV"
        href="/tokens"
        onClick={[Function]}
      >
        Create eToken
      </a>
      <p>
        Tokens sent to your 
        eToken
         address will appear here
      </p>
    </div>
  </div>,
]
`;

exports[`Wallet with BCH balances and tokens and state field 1`] = `
Array [
  <div
    className="sc-elJkPf KqQMk"
  >
    <div
      className="sc-eXEjpC fRnTPw"
    >
      <h4
        className="sc-ibxdXY gNKGGy"
      >
        MigrationTestAlpha
      </h4>
      <a
        href="/configure"
        onClick={[Function]}
      >
        <svg
          className="sc-jTzLTM bTdWCF"
        >
          edit.svg
        </svg>
      </a>
      <div>
         
        <button
          aria-checked={false}
          className="ant-switch ant-switch-small"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="switch"
          type="button"
        >
          <div
            className="ant-switch-handle"
          />
          <span
            className="ant-switch-inner"
          >
            <svg
              className="sc-kAzzGY fUTkYj"
            >
              eye-invisible.svg
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      className="sc-kTUwUJ bWNEPO"
    >
      <span
        className="sc-kfGgVZ ezlXcw"
      >
         
        0.06
         
        XEC
         
      </span>
    </div>
  </div>,
  <div
    className="sc-lhVmIH gJoLAm"
  >
    <div
      className="sc-RefOD bFUtiM"
    >
      <button
        className="sc-iQKALj bpbBVm"
        onClick={[Function]}
      >
        Transactions
      </button>
      <button
        className="sc-iQKALj jssRHA"
        onClick={[Function]}
      >
        eTokens
      </button>
    </div>
    <div
      className="sc-bwCtUz jWzyRI"
    >
      <div />
      <span
        aria-label="party emoji"
        role="img"
      >
        ðŸŽ‰
      </span>
      Congratulations on your new wallet!
       
      <span
        aria-label="party emoji"
        role="img"
      >
        ðŸŽ‰
      </span>
      <br />
       Start using the wallet immediately to receive
       
      XEC
       payments, or load it up with
       
      XEC
       to send to others
      <br />
      <br />
    </div>
    <div
      className="sc-bwCtUz hyFPDq"
    >
      <a
        className="sc-hXRMBi kpUYV"
        href="/tokens"
        onClick={[Function]}
      >
        Create eToken
      </a>
      <div>
        <a
          href="/send-token/bd1acc4c986de57af8d6d2a64aecad8c30ee80f37ae9d066d758923732ddc9ba"
          onClick={[Function]}
        >
          <div
            className="sc-jlyJG gcRGZa"
          >
            <div
              className="sc-iRbamj ckvBio"
            >
              <div
                className="sc-gPEVay jrVrPT"
              />
              <h4>
                TBS
              </h4>
            </div>
            <h4>
              6.001000000
            </h4>
          </div>
        </a>
      </div>
    </div>
  </div>,
]
`;

exports[`Wallet without BCH balance 1`] = `
Array [
  <div
    className="sc-elJkPf KqQMk"
  >
    <div
      className="sc-eXEjpC fRnTPw"
    >
      <h4
        className="sc-ibxdXY gNKGGy"
      >
        MigrationTestAlpha
      </h4>
      <a
        href="/configure"
        onClick={[Function]}
      >
        <svg
          className="sc-jTzLTM bTdWCF"
        >
          edit.svg
        </svg>
      </a>
      <div>
         
        <button
          aria-checked={false}
          className="ant-switch ant-switch-small"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="switch"
          type="button"
        >
          <div
            className="ant-switch-handle"
          />
          <span
            className="ant-switch-inner"
          >
            <svg
              className="sc-kAzzGY fUTkYj"
            >
              eye-invisible.svg
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      className="sc-kTUwUJ bWNEPO"
    >
      <span
        className="sc-kfGgVZ ezlXcw"
      >
         
        0
         
        XEC
         
      </span>
    </div>
  </div>,
  <div
    className="sc-lhVmIH gJoLAm"
  >
    <div
      className="sc-RefOD bFUtiM"
    >
      <button
        className="sc-iQKALj bpbBVm"
        onClick={[Function]}
      >
        Transactions
      </button>
      <button
        className="sc-iQKALj jssRHA"
        onClick={[Function]}
      >
        eTokens
      </button>
    </div>
    <div
      className="sc-bwCtUz jWzyRI"
    >
      <div />
      <span
        aria-label="party emoji"
        role="img"
      >
        ðŸŽ‰
      </span>
      Congratulations on your new wallet!
       
      <span
        aria-label="party emoji"
        role="img"
      >
        ðŸŽ‰
      </span>
      <br />
       Start using the wallet immediately to receive
       
      XEC
       payments, or load it up with
       
      XEC
       to send to others
      <br />
      <br />
    </div>
    <div
      className="sc-bwCtUz hyFPDq"
    >
      <a
        className="sc-hXRMBi kpUYV"
        href="/tokens"
        onClick={[Function]}
      >
        Create eToken
      </a>
      <p>
        Tokens sent to your 
        eToken
         address will appear here
      </p>
    </div>
  </div>,
]
`;

exports[`Without wallet defined 1`] = `
<div
  className="sc-cMljjf yzuNU"
>
  <h2>
    Welcome to Cashtab!
  </h2>
  <p
    className="sc-jAaTju dRNmSH"
  >
    Cashtab is an
     
    <a
      className="sc-jDwBTQ cKENCL"
      href="https://github.com/bitcoin-abc/bitcoin-abc"
      rel="noreferrer"
      target="_blank"
    >
      open source,
    </a>
     
    non-custodial web wallet for 
    eCash
    .
  </p>
  <button
    className="sc-eHgmQL bqOmSo"
    onClick={[Function]}
  >
    <span
      aria-label="plus-square"
      className="anticon anticon-plus-square"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="plus-square"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M328 544h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
        />
        <path
          d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
        />
      </svg>
    </span>
     New Wallet
  </button>
  <button
    className="sc-cvbbAY eOLbDk"
    onClick={[Function]}
  >
    <span
      aria-label="import"
      className="anticon anticon-import"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="import"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M888.3 757.4h-53.8c-4.2 0-7.7 3.5-7.7 7.7v61.8H197.1V197.1h629.8v61.8c0 4.2 3.5 7.7 7.7 7.7h53.8c4.2 0 7.7-3.4 7.7-7.7V158.7c0-17-13.7-30.7-30.7-30.7H158.7c-17 0-30.7 13.7-30.7 30.7v706.6c0 17 13.7 30.7 30.7 30.7h706.6c17 0 30.7-13.7 30.7-30.7V765.1c0-4.3-3.5-7.7-7.7-7.7zM902 476H588v-76c0-6.7-7.8-10.5-13-6.3l-141.9 112a8 8 0 000 12.6l141.9 112c5.3 4.2 13 .4 13-6.3v-76h314c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8z"
        />
      </svg>
    </span>
     Import Wallet
  </button>
</div>
`;
